### ISSUE TITLE
error: cross-crate traits with a default impl, like `Sync`, should not be specialized

### ISSUE THREAD

**Issue by starkoracles** [rkyv/rend#7]
When running build on this crate with 
```
➜  rkyv git:(master) ✗ cargo --version
cargo 1.63.0 (fd9c4297c 2022-07-01)
➜  rkyv git:(master) ✗ rustc --version
rustc 1.63.0 (4b91a6ea7 2022-08-08)
➜  rkyv git:(master) ✗ 
```

I'm getting 
```
error: cross-crate traits with a default impl, like `Sync`, should not be specialized
   --> /Users/dev/rend/src/impl_struct.rs:396:9
    |
396 |           unsafe impl Sync for $endian<$ne> {}
    |           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
   ::: /Users/dev/rend/src/lib.rs:338:1
    |
338 | / impl_endian!(
339 | |     @atomic AtomicU64 = u64,
340 | |     AtomicU64_ne = "`NativeEndian<AtomicU64>`",
341 | |     AtomicU64_le = "`LittleEndian<AtomicU64>`",
342 | |     AtomicU64_be = "`BigEndian<AtomicU64>`"
343 | | );
    | |_- in this macro invocation
    |
    = warning: this will change its meaning in a future release!
    = note: for more information, see issue #93367 <https://github.com/rust-lang/rust/issues/93367>
    = note: `AtomicU64` is not a generic parameter
note: try using the same sequence of generic parameters as the struct definition
   --> /Users/dev/rend/src/lib.rs:110:1
    |
110 | / pub struct BigEndian<T> {
111 | |     value: T,
112 | | }
    | |_^
    = note: this error originates in the macro `impl_struct` (in Nightly builds, run with -Z macro-backtrace for more info)

error: could not compile `rend` due to 36 previous errors
```

would love to hear if you have thoughts why?

**Comment by djkoloski** [comment]
The issue discussing the change is a little dense but these impls are unnecessary since we can get the default impl automatically. FIxed by 2f0d5046666ab96114046c1ad970510b2a7febe3

**Comment by starkoracles** [comment]
Thanks!
